Update seer docs (#14027)
getsentry/sentry-docs: docs/product/ai-in-sentry/seer/issue-fix/index.mdx

---
title: Issue Fix
description: "Use Seer's issue fix to automatically find the root cause of issues and generate code fixes."
sidebar_order: 20
---

**Issue fix** is Seer's collaborative workflow to find the root cause of and solution to issues. It uses Sentry's context (issue details, tracing data, logs, and profiles), your codebases (integrated through GitHub), and its interactions with you to identify and reason through problems in your code.

This tool can be automated, allowing it to populate root cause analysis, solutions, and even Pull Requests as issues occur.

<Alert>

> Issue Fix builds upon the capabilities previously known as Autofix.

Reviewer: ```suggestion
Issue fix builds upon the capabilities previously known as Autofix.
```

1. [Root Cause Analysis](#root-cause-analysis): Analyze the issue using Sentry's context and your codebases and determine the root cause
2. [Solution Identification](#solution-identification): Identify a solution based on the analysis
3. [Code Generation](#code-generation): Generate a code fix and open a PR

When you begun debugging an issue you Seer, you will see a screen like this, which includes the high-level issue summary and a button to start the issue fix flow:

![Issue Fix Start Screen](../img/issue-fix-start-screen.png)

### Root Cause Analysis

Seer will then analyze the issue and relevant code to help identify the root cause and a solution. You'll see a live stream of Seer's thought process in real time as it works through the problem. Seer will share key insights as they come up and you'll have the option to expand each. You'll see the event data and code that support Seer's conclusions, building a clear chain of reasoning back to the root cause.

While Seer is reasoning through your problem, you can step in to add context or feedback; collaboration tends to lead to the best results. If Seer encounters gaps in its understanding, it will ask for your input.

> The final root cause analysis will show you the sequence of events that led to the issue, and a detailed analysis of the most likely culprit. This analysis includes helpful annotations, such as links back to the exact lines of code or sentry telemetry that are relevant to the analysis.

Reviewer 2: ```suggestion
The final root cause analysis will show you the sequence of events that led to the issue, and a detailed analysis of the most likely culprit. This analysis includes helpful annotations, such as links back to the exact lines of code or Sentry telemetry that are relevant to the analysis.
```

- **Performance data**: Profiles and performance metrics
- **Interactive feedback**: Your input and guidance during the process

This context is crucial to Seer's capabilities, and it allows it to do things like investigate a backend service that may be the true root cause for a frontend bug you are trying to troubleshoot. 

## Settings

Issue fix's default behavior can be customized in several ways:

1. You can manually customize the repositories it has access to
2. You can define the default branch it should use in each repository
3. You can add any additional custom context that that may be useful within the scope of each individual repository

<Alert>
Seer already automatically parses rules files from [Cursor](https://docs.cursor.com/context/rules), [Windsurf](https://windsurf.com/editor/directory), [Cline](https://docs.cline.bot/features/cline-rules), and [Claude Code](https://docs.anthropic.com/en/docs/claude-code/tutorials#create-an-effective-claude-md-file)
</Alert>

![Screenshot of the repository settings page](../img/repo-settings.png)

## Best Practices

> - Setup the GitHub integration, allowing Seer access to your codebase and enabling it to generate code fixes. Enabling the GitHub integration will make Seer much smarter and more capable. See the [Sentry GitHub integration](/organization/integrations/source-code-mgmt/github/) to learn how

Reviewer 2: I'm not sure about saying it will make Seer 'smarter', would it make sense to say it provides it with more context or something like that instead?

Reviewer: ```suggestion
- Setup the GitHub integration, allowing Seer access to your codebase and enabling it to generate code fixes. Enabling the GitHub integration will make Seer provide better results. See the [Sentry GitHub integration](/organization/integrations/source-code-mgmt/github/) to learn how
```

How about this?

![Coding Step](../img/coding-step.png)

## What Issue Fix Uses

Seer is a powerful debugging agent, with access to a wide variety of data sources and tools. While debugging issues, it may examine:

- **Issue Context**: Error messages, stack traces, and event metadata from your [Issues](/product/issues/)
- **Tracing Data**: Distributed [traces](/concepts/key-terms/tracing/#whats-a-trace) and span information
- **Logs**: Structured [Logs](/product/explore/logs/) from your application (beta)
- **Your Codebase**: Relevant code from linked GitHub repositories, with support for multiple repos for distributed services
- **Performance Data**: Profiles and performance metrics
- **Interactive Feedback**: Your input and guidance during the process

This context is crucial to Seer's capabilities, and it allows it to do things like investigate a backend service that may be the true root cause for a frontend bug you are trying to troubleshoot. 

## Settings

Issue Fix's default behavior can be customized in several ways:

> 1. You can manually customize the repositories it has access to. 

Reviewer 3: ```suggestion
1. You can manually customize the repositories it has access to
```

2. You can define the default branch it should use in each repository
3. You can add any additional custom context that that may be useful within the scope of each individual repository
<Alert>
Seer already automatically parses rules files from [Cursor](https://docs.cursor.com/context/rules), [Windsurf](https://windsurf.com/editor/directory), [Cline](https://docs.cline.bot/features/cline-rules) 

> </Alert>

Reviewer 3: should call out Claude Code/CLAUDE.md as well

![Repository Settings](../img/repo-settings.png)

## Prerequisites

To use Issue Fix, you'll need to have the following:

> 1. A Sentry project with issues. The more context you have (stacktrace, tracing data, logs, and profiles), the better equipped Seer will be to help. 

Reviewer 3: ```suggestion
1. A Sentry project with issues. The more context you have (stack traces, traces, logs, profiles, and more), the better equipped Seer will be to help. 
```

> 2. Working knowledge of the issue you want to fix. Seer will help you identify the root cause and suggest a solution, but you should have a general understanding of the problem you're trying to solve.

Reviewer 3: what's this mean?

Author: will rewrite this section

> You will need a GitHub integration set up in your Sentry project in order for Seer to access your codebase and generate code fixes. While Seer will work without this, enabling the GitHub integration will make Seer much smarter and more capable. See the [Sentry GitHub integration](/organization/integrations/source-code-mgmt/github/) to learn more about this process. 

Reviewer 3: should call out that other SCMs are not supported and are coming soon
